Add add_multiline_prefix and blockquoter

dpprdan opened the pull request:
This adds a `blockquoter` function, almost identical to the `listr` function. Inspired by the `add_prefix` function I pulled out the main functionality into the `add_multiline_prefix` function. 

I added tests for `blockquoter` (again copying from `listr`), but most of the tests don't run for me because they use an `it` function, which is not defined nor imported, as far as I can tell.

Let me know, if this fits the scope of the package.

yonicd commented:
resolve conflicts and i'll accept

yonicd commented:
i think this was taken care of by @aoles's PR.

aoles commented:
I think this actually is a new feature different from the multiple selections support which I introduced recently. If I understand it correctly @dpprdan functions operate on a single selection but process each line separately to add the prefix.

dpprdan commented:
@aoles exactly! 😃 @aoles functions `add_prefix` and `enclose` support multiple single-line selections (i.e. cursors), my `add_multiline_prefix` supports one multi-line selection. 

Actually I'd prefer RStudio's "Comment/Uncomment Lines" (Ctrl/CMD+Shift+C) behaviour, in particular:

- select whole line even if only partially selected (including selecting the current line when nothing is selected)
- respect indentation 
  - (handy for lists with sub-items like this one)
- ability to uncomment (unlist, unblockquote)

Do you know whether that's possible with {rstudioapi}, or is that a built-in-only functionality of RStudio?

aoles commented:
@dpprdan AFAIK there is no such functionality provided in rstudioapi, so you would need to implement the toggle behavior such as comment-uncomment yourself.

dpprdan commented:
I submitted some changes:

1. resolved conflicts.
2. added argument `as_is` to `add_multiline_prefix`: In order to support multiple paragraphs in a single blockquote, blank lines must be prefixed as well. This is done by `as_is = TRUE`. `as_is = FALSE` now also supports respects indentation for nested lists.
3. added `olistr` for ordered lists.

yonicd commented:
looks good. are there tests for listr and olistr?

yonicd commented:
also since tests are skipped on travis please run covrpage and update the readme in the tests folder.

dpprdan commented:
Well, there were some tests for `listr` and `backquoter`, but not up-to-date, and no tests for `add_multiline_prefix` and `olistr`. So I added those. Furthermore a fix so `add_multiline_prefix` does not add a linebreak on an empty selection. I also added a `Sys.sleep(.1)` to set_text, since I got incosistent test results. See this [gist](https://gist.github.com/dpprdan/041c1d557c046dfa4f0253bddf90e9fc) for example where I ran a couple of `devtools::test(filter = "multi|block|list")` shortly after one another. After that, `covrpage` also finally worked for me (had some failing tests from functions I did not touch, e.g. `chunkr` or `backtickr` before). Finally I took the liberty to add myself as contributor, I hope that's okay.

yonicd commented on tests/testthat/helper-functions.R:
```diff
@@ -38,7 +38,7 @@ individual_lines <- function() {
   rstudioapi::modifyRange(location = entire_document(), text = txt, id = sec$id)
   rstudioapi::documentSave(sec$id) 
-  
+  Sys.sleep(.1)
```
i also had this in here originally, didnt notice it was removed.

dpprdan commented:
👍
